src/medCore/gui/toolboxes/medToolBox.cpp

```diff
@@ -308,3 +308,39 @@ void medToolBox::toXMLNode(QDomDocument* doc, QDomElement* currentNode)
 	currentNode->appendChild(elmt);
 }
 
+void medToolBox::handleDisplayError(int error)
+{
+    switch (error)
+    {
+    case medAbstractProcess::PIXEL_TYPE:
+        displayPixelTypeError();
+        break;
+    case medAbstractProcess::DIMENSION:
+        displayDataDimensionError();
+        break;
+    case medAbstractProcess::MESH_TYPE:
+        displayMeshTypeError();
+        break;
+    }
+}
+
+void medToolBox::displayPixelTypeError()
+{
+    QString error = "Pixel type not yet implemented";
+    qDebug() << name() + ": " + error;
+    medMessageController::instance()->showError(error,3000);
+}
+
+void medToolBox::displayDataDimensionError()
+{
+    QString error = "This toolbox is designed to be used with 3D volumes";
+    qDebug() << name() + ": " + error;
+    medMessageController::instance()->showError(error,3000);
+}
+
+void medToolBox::displayMeshTypeError()
```

Comment:
Ok I agree that predefined messages is good, so leave ```handleDisplayError```, but it can use a ```displayError```function that the user would also be free to use.